sdlib/d/gc/page.d

```diff
@@ -726,9 +726,38 @@ private:
 					auto newOccupancy = oldOccupancy & bmp[i];
 
 					occupancyMask |= newOccupancy;
-					count += popCount(oldOccupancy ^ newOccupancy);
+					auto toRemove = oldOccupancy ^ newOccupancy;
+					count += popCount(toRemove);
 
-					e.slabData.rawContent[i] = newOccupancy;
+					scope(exit) e.slabData.rawContent[i] = newOccupancy;
```

Comment:
scope(success) no? There is a real question fo what to do when the finalizer throw, but I think we are fine ignoring this for now.

Reply:
I don't know what happens now. But certainly I can change this to scope(success).